homework2/main.py

```diff
@@ -0,0 +1,170 @@
+
+
+class MovieDataGrabber:
+    def __init__(self, num_pages=1):
+        self.api_key = API_KEY
+        self.num_pages = num_pages
+        self.url = 'https://api.themoviedb.org/3/'
+        self.data = []
+        self.genres_data = None
+        self.headers = {
+            'accept': 'application/json',
+            'Authorization': f'Bearer {self.api_key}'
+        }
+
+        self.params = {
+            'include_adult': False,
+            'include_video': False,
+            'sort_by': 'popularity.desc'
+        }
+        self.fetch_data()
+        self.fetch_genres_data()
+
+    def fetch_data(self):
+        for page_num in range(1, self.num_pages + 1):
+            self.params['page'] = page_num
+            response = requests.get(f'{self.url}discover/movie', params=self.params, headers=self.headers)
+            if response.status_code == 200:
+                data = response.json()
+                self.data.extend(data.get('results', []))
+            else:
+                print(f'Error fetching data from page {page_num}')
+
+    def fetch_genres_data(self):
+        try:
+            response = requests.get(f'{self.url}genre/movie/list?language=en', headers=self.headers)
+            response.raise_for_status()
+
+            genres_data = response.json()
+            self.genres_data = genres_data.get('genres', [])
+
+        except requests.exceptions.RequestException as e:
+            print(f'Error fetching genres data: {e}')
+
+    def get_all_data(self):
+        return self.data
+
+    def get_genres_data(self):
+        return self.genres_data
+
+    def get_data_by_indexes(self, start_index, end_index, step):
+        return self.data[start_index:end_index:step]
+
+    def get_most_popular_title(self):
+        return max(self.data, key=lambda x: x['popularity']).get('title', '')
+
+    def get_titles_by_description_key_words(self, keywords):
+        return [movie.get('title', '') for movie in self.data if
+                any(keyword in movie.get('overview', '').lower() for keyword in keywords.lower().split())]
+
+    def get_unique_genres(self):
+        return (genre['name'] for genre in self.genres_data)
+
+    def delete_movie_by_genre(self, genre):
+        # genre_id = (item['id'] for item in self.genres_data if item['name'] == genre)[0]
+        genre_id = [item['id'] for item in self.genres_data if item['name'] == genre].pop()
+        self.data = [movie for movie in self.data if genre_id not in movie.get('genre_ids', [])]
+
+    def names_of_most_popular_genres(self, count=1):
+        all_genre_ids = [genre_id for movie in self.data for genre_id in movie.get('genre_ids', [])]
+        most_common_genre_ids = Counter(all_genre_ids).most_common(count)
+        return [(next(item['name'] for item in self.genres_data if item['id'] == id_val), score_val)
+                for id_val, score_val in most_common_genre_ids]
```

Comment:
I think we alredy discussed with you that using **next()** is a bad way of receiving the only value